src/sign1.rs

```diff
@@ -173,6 +176,17 @@ impl CoseSign1 {
     pub fn payload(&self) -> Option<&ByteBuf> {
         self.inner.2.as_ref()
     }
+
+    pub fn claims_set(&self) -> Result<Option<ClaimsSet>> {
```

Comment:
is there a more idiomatic (or otherwise preferred) way of handling this type of case? Result<Option<> feels a bit clunky but makes it explicit (instead of adding two errors for empty vs. failure)

Reply:
`Result<Option<>>` is right

Reply:
oops didn't see this til now 🙈 as a minor note, I switched to `Option<Result<ClaimsSet>>` to match your linked example oid4vp + get_claims, and since the code seemed a tad simpler -- happy to change back if needed!